blocks/file-blocks/use-case/index.tsx

```diff
@@ -0,0 +1,175 @@
+
+  const getBlocks = async () => {
+    const blocksRepos = await onRequestBlocksRepos();
+    let blocks = {};
+    blocksRepos.forEach((repo) => {
+      repo.blocks.forEach((block) => {
+        const id = [repo.owner, repo.repo, block.id].join("__");
+        blocks[id] = block;
+      });
+    });
+    setBlockOptions(blocks);
+  };
+  useEffect(() => {
+    getBlocks();
+  }, []);
+
+  const components = useMemo(
+    () => ({
+      BlockComponent: BlockComponent({ ...props, blockOptions }),
+      code({
+        inline,
+        className,
+        children,
+      }: {
+        inline: boolean;
+        className: string;
+        children: any;
+      }) {
+        const match = /language-(\w+)/.exec(className || "");
+        return !inline && match ? (
+          <div className={tw(`code`)}>
+            <SyntaxHighlighter language={match[1]}>
+              {String(children).replace(/\n$/, "")}
+            </SyntaxHighlighter>
+          </div>
+        ) : (
+          <code className={className}>{children}</code>
+        );
+      },
+      a: InlineLink,
+    }),
+    [BlockComponent, blockOptions]
+  );
+
+  return (
+    <div className={tw(`w-full h-full flex items-stretch overflow-hidden`)}>
+      <div
+        className={tw(
+          `flex-1 markdown-body p-6 pb-40 overflow-y-auto whitespace-pre-wrap`
+        )}
+      >
+        <div className={tw(`max-w-[60em] mx-auto`)}>
+          <ErrorBoundary key={content}>
+            <MDX components={components}>{content}</MDX>
+          </ErrorBoundary>
+        </div>
+      </div>
+    </div>
+  );
+}
+function InlineLink(props: Record<string, any>) {
+  const videoExtensions =
+    /\.(mp4|webm|ogv|mov|flv|wmv|avi|m4v|mpg|mpeg|3gp|3g2)$/i;
+  const isVideo = videoExtensions.test(props.href);
+
+  if (isVideo)
+    return (
+      <video controls>
+        <source src={props.href} />
+      </video>
+    );
+
+  return (
+    <a target="_blank" rel="noreferrer" {...props}>
+      {props.children}
+    </a>
+  );
+}
+
+const BlockComponent =
+  (
+    props: FileBlockProps & {
+      blockOptions: Record<string, Block>;
+    }
+  ) =>
+  ({ blockId = "", path = "", context = {} }) => {
+    const { BlockComponent, blockOptions } = props;
+
+    const block = blockOptions[blockId];
+
+    const combinedContext = useMemo(
+      () => ({
+        ...props.context,
+        ...context,
+      }),
+      [props.context, context]
+    );
+
+    return (
+      <div className={tw(`mt-3 mb-6`)}>
+        <Box
+          className={tw(
+            `flex items-center justify-between py-3 px-5 bg-slate-100`
+          )}
+          bg="canvas.subtle"
+        >
+          <div>
+            <strong>{block?.title || blockId}</strong> showing
+            <code className="!ml-1 inline-block">
+              {combinedContext.owner}/{combinedContext.repo}
+            </code>
+            :<code className="!ml-1 inline-block">{path || "/"}</code>
+          </div>
+          <Link
```

Comment:
I was going to say we could use `onNavigateToPath` until I noticed we're passing a block ID. Maybe this should be part of `onNavigateToPath`'s interface? (Fine for this PR.)